server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -85,19 +113,54 @@ public Builder setStored(boolean value) {
 
         @Override
         protected Parameter<?>[] getParameters() {
-            return new Parameter<?>[] { stored, meta };
+            return new Parameter<?>[] { stored, meta, indexOptions };
         }
 
         @Override
         public SparseVectorFieldMapper build(MapperBuilderContext context) {
             return new SparseVectorFieldMapper(
                 leafName(),
-                new SparseVectorFieldType(context.buildFullName(leafName()), stored.getValue(), meta.getValue()),
+                new SparseVectorFieldType(context.buildFullName(leafName()), stored.getValue(), meta.getValue(), indexOptions.getValue()),
                 builderParams(this, context)
             );
         }
     }
 
+    public IndexOptions getIndexOptions() {
+        return fieldType().getIndexOptions();
+    }
+
+    private static final ConstructingObjectParser<IndexOptions, Void> INDEX_OPTIONS_PARSER = new ConstructingObjectParser<>(
+        SPARSE_VECTOR_INDEX_OPTIONS,
+        args -> new IndexOptions((Boolean) args[0], (TokenPruningConfig) args[1])
```

Comment:
I don't know why we are using a nullable boolean here. Let's default appropriately and be clear about the configuration in the mapping as we are changing default behavior for users.